web/app/scripts/custom-reports/custom-reports-modal-controller.js

```diff
@@ -153,16 +90,23 @@
                         delete params.limit;
                     }
 
-                    // We now have all the parameters needed for performing a crosstabs query.
-                    // TODO: when reports page exists, open it in a new window with these params.
-
-                    /* jshint camelcase: true */
-
-                    $modalInstance.close();
+                    ctl.params = params;
                 }
             );
         }
 
+        function createReport() {
+            // If we were using $resource, Angular would automagically encode params that are
+            // objects. But since we're using $state.href, we have to do it ourselves.
+            _.forEach(ctl.params, function (value, key) {
```

Comment:
Consider creating a new object to pass to `$state.href` so that you're not mutating the state of the existing ctl.params object. Certainly doesn't matter now but could be a nasty side effect in the future if more stuff is added to this controller.